force-app/main/default/classes/DeceasedBatch.cls

```diff
@@ -0,0 +1,167 @@
+                allMembersDeceasedById.put(householdMember.AccountId, householdMember.Account.All_Members_Deceased__c);
+            }
+        }
+
+        if (highLowProcessedIds.contains(scope[0].AccountId)) {
+            allMembersDeceasedById.remove(scope[0].AccountId);
+        }
+        if (highLowProcessedIds.contains(scope[scope.size() - 1].AccountId)) {
+            allMembersDeceasedById.remove(scope[scope.size() - 1].AccountId);
+        }
+
+        return allMembersDeceasedById;
+    }
+
+    private List<Contact> getContacts(Set<Id> accountIds) {
+        List<Contact> contacts = [
+            SELECT AccountId, Deceased__c
+            FROM Contact
+            WHERE AccountId IN :accountIds
+            ORDER BY AccountId
+        ];
+        return contacts;
+    }
+
+    @TestVisible
+    private List<Account> getAllMembersDeceasedUpdates(List<Contact> contacts, Map<Id, Boolean> allMembersDeceasedById) {
+        Id previousAccount;
+        Boolean currentAllMembersDeceasedValue;
+        Boolean calculatedDeceasedValue;
+        List<Account> deceasedUpdates = new List<Account>();
+
+        for (Contact householdMember : contacts) {
+            if (!allMembersDeceasedById.containsKey(householdMember.AccountId)) {
+                continue;
+            }
+
+            if (previousAccount != householdMember.AccountId) {
+                if (currentAllMembersDeceasedValue != calculatedDeceasedValue) {
+                    deceasedUpdates.add(newUpdatedAccount(previousAccount, calculatedDeceasedValue));
+                }
+                previousAccount = householdMember.AccountId;
+                currentAllMembersDeceasedValue = allMembersDeceasedById.get(householdMember.AccountId);
+                calculatedDeceasedValue = true;
+            }
+
+            if (householdMember.Deceased__c != true) {
+                calculatedDeceasedValue = false;
+            }
+        }
+
+        if (currentAllMembersDeceasedValue != calculatedDeceasedValue) {
+            deceasedUpdates.add(newUpdatedAccount(previousAccount, calculatedDeceasedValue));
+        }
+
+        return deceasedUpdates;
+    }
+
+    private Account newUpdatedAccount(Id previousAccountId, Boolean calculatedAllMembersDeceasedSetting) {
+        return new Account(
+            Id = previousAccountId,
+            All_Members_Deceased__c = calculatedAllMembersDeceasedSetting
+        );
+    }
+
+    private void updateModifiedAccounts(List<Account> allMembersDeceasedUpdates) {
+        TDTM_Config_API.disableAllTriggers();
+        Database.update(allMembersDeceasedUpdates, false);
+    }
+
+    @TestVisible
+    private void saveHighAndLowProcessedIds(List<Contact> contacts) {
+        highLowProcessedIds.add(contacts[0].AccountId);
+        highLowProcessedIds.add(contacts[contacts.size() - 1].AccountId);
+    }
+
+    public void finish(Database.BatchableContext info) {
```

Comment:
This may not be something we care about enough to handle.  Is this worth doing???

Reply:
Ehhh, the Deceased flag gets unchecked when the last Contact gets removed (must be in the Trigger), so this is definitely an edge case. That being said, it's already here, other than one extra query it can't really hurt.